### ISSUE TITLE
Warnings about authentication storage backends

### ISSUE THREAD

**Issue by benmoss** [prefix-dev/rattler-build#589]
I'm not sure what these warnings are about, but I see a million of them in my builds:

```
2024-02-02T20:44:40.359834Z  WARN rattler_networking::authentication_storage::storage: Error retrieving credentials from backend: Platform secure storage failure: no secret service provider or dbus session found
2024-02-02T20:44:40.360430Z  WARN rattler_networking::authentication_storage::backends::file: Can't find path for file storage on /root/.rattler/credentials.json
```

**Comment by orhun** [comment]
I also get these :/ I can probably look into it.

**Comment by orhun** [comment]
TL;DR: This error occurs if there's no secret service daemon (e.g. KWallet, gnome-keyring, KeePassXC, ...) running. [\*](https://github.com/hwchen/secret-service-rs/issues/51)

`rattler-build` depends on the following authentication storage backends from `rattler_networking`:

- `file`
- `keyring`
- `netrtc`

And this error originates from the `keyring` implementation. We depend on the [`keyring`](https://docs.rs/keyring) crate for storing credentials in a compatible daemon such as KWallet, gnome-keyring or KeePassXC.

You can simply reproduce the error with this snippet:

```rs
// [dependencies]
// keyring = "2.3.2"

use keyring::Entry;

fn main() {
    let entry = Entry::new("my_service", "my_name").unwrap();
    entry.set_password("topS3cr3tP4$$w0rd").unwrap();
}
```

> thread 'main' panicked at src/main.rs:5:45:
called `Result::unwrap()` on an `Err` value: PlatformFailure(Zbus(MethodError(OwnedErrorName(ErrorName(Str(Owned("org.freedesktop.DBus.Error.ServiceUnknown")))), Some("The name is not activatable"), Msg { type: Error, sender: UniqueName(Str
(Borrowed("org.freedesktop.DBus"))), reply-serial: 2, body: Signature("s") })))
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

`keyring` crate also supports multiple backends:

```toml
[target.'cfg(target_os = "linux")'.dependencies]
secret-service = { version = "3", optional = true }
linux-keyutils = { version = "0.2", features = ["std"], optional = true }
```

[`secret-service`](https://docs.rs/secret-service/) crate is used as default which depends on a secret service running.

>  Secret Service provides a secure place to store secrets. Gnome keyring and KWallet implement the Secret Service API.

Also, see <https://specifications.freedesktop.org/secret-service/latest/>

Which also means we can reproduce the same error with this snippet:

```rs
// [dependencies]
// secret-service = { version = "3.0.0", features = ["rt-tokio-crypto-rust"] }
// tokio = { version = "1.36.0", features = ["full"] }

use secret_service::EncryptionType;
use secret_service::SecretService;
use std::error::Error;

#[tokio::main]
async fn main() -> Result<(), Box<dyn Error>> {
    // initialize secret service (dbus connection and encryption session)
    let ss = SecretService::connect(EncryptionType::Dh).await.unwrap();

    Ok(())
}
```

> thread 'main' panicked at src/main.rs:15:63:
called `Result::unwrap()` on an `Err` value: Zbus(MethodError(OwnedErrorName(ErrorName(Str(Owned("org.freedesktop.DBus.Error.ServiceUnknown")))), Some("The name is not activatable"), Msg { type: Error, sender: UniqueName(Str(Borrowed("org.f
reedesktop.DBus"))), reply-serial: 2, body: Signature("s") }))
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

There is also another case where a secret service is available but it is **locked**. In that case, we need to unlock it as follows:

> If you are running on a headless Linux box, you will need to unlock the Gnome login keyring before you can use it.  The following `bash` function may be very helpful.

```shell
function unlock-keyring ()
{
    read -rsp "Password: " pass
    echo -n "$pass" | gnome-keyring-daemon --unlock
    unset pass
}
```

> Trying to access a locked keychain on a headless Linux box often returns the  platform error that displays as `SS error: prompt dismissed`.  This refers to the fact that there is no GUI running that can be used to prompt for a keychain unlock.

---

Okay so, what can we do?

Luckily, `keyring` crate [supports](https://github.com/hwchen/keyring-rs/issues/153) another backend (`linux-keyutils`) which will work on Linux systems without Gnome keyring etc.

```rs
// keyring = { version = "2.3.2", default-features = false, features = ["linux-keyutils"] }

use keyring::Entry;

fn main() {
    let entry = Entry::new("my_service", "my_name").unwrap();
    entry.set_password("topS3cr3tP4$$w0rd").unwrap();
}
```

The only downside of this approach is the keyring is provided in-memory by the running kernel so it will not persist across reboots.

See <https://docs.rs/keyring/latest/keyring/keyutils/index.html> for more information.

With all of this considered, it might be a good idea to switch to `linux-keyutils` backend for authentication storage.

**Comment by wolfv** [comment]
We do have a fallback storage implemented that stores authentication data in a JSON file and which users can use if they don't want to run a fancy authentication daemon. 

However, we need to silently fall through and ignore this warning / error.

**Comment by orhun** [comment]
I see, I created #599 to suppress the warnings.
